crates/sui-core/src/consensus_adapter.rs

```diff
@@ -850,6 +869,25 @@ impl ConsensusAdapter {
                         )
                         .await;
 
+                    if let Some(submitted_cache) = &epoch_store.submitted_transaction_cache {
+                        for (transaction, consensus_position) in
+                            transactions.iter().zip(consensus_positions.iter())
+                        {
+                            if let ConsensusTransactionKind::UserTransaction(tx) = &transaction.kind
+                            {
+                                let submitted_round = consensus_position.block.round;
+                                let digest = tx.digest();
+
+                                submitted_cache.record_submitted_tx(
+                                    digest,
+                                    submitted_round,
+                                    amplification_factor as u32,
```

Comment:
1. It seems if we use a LRU, we don't need to record the transaction until consensus adapter. It seems better to record submission early too.
2. For the amplification factor used by the submission cache, I think we can just use
```
(gas_price / epoch_store.reference_gas_price().max(1)).max(1)
```
without `sip_45_consensus_amplification_threshold` or adding another `1`  amplification later.

Reply:
I believe the reason to record submission after the txn is included in a block, is to obtain the round number to evict txns. If the eviction policy is pure LRU based, I think txn submissions can be recorded pretty early in ConsensusAdapter. I may be missing something though.